lib/search_solr_tools/logging/sst_logger.rb

```diff
@@ -0,0 +1,76 @@
+# frozen_string_literal: true
+
+require 'logging'
+require 'search_solr_tools'
+
+module SSTLogger
+  LOG_LEVELS = %w[debug info warn error fatal none].freeze
+
+  def logger
+    SSTLogger.logger
+  end
+
+  def log_environment(new_env)
+    SSTLogger.log_environment(new_env)
+  end
+
+  class << self
+    def logger
+      @logger ||= new_logger
+    end
+
+    def log_environment(new_env)
```

Comment:
I was trying to avoid puts statements, as then you have to add manual flags to each statement.  By using the logger to stdout, you can just do "logger.info" and it will potentially write to one or both places automatically.  No separate checking needed.

That said, I could have a verbose flag set up the stdout log, as opposed to having it be a configuration/environment variable, but that would mean having to pass that verbosity into the logger initialization.

Reply:
I should also note that the output for the stdout and the log file are slightly different - the stdout console doesn't put datetime stamps, whereas those ARE written to the log files.

Reply:
Also, is this comment in the right place? This log_environment method is just to set the environment passed into the CLI (integration, staging, etc), as the logger module would not have that information otherwise.

Reply:
I suppose I could add the comment at line 13, but the gist would be the same. The name of the method gives me the wrong idea about what it's doing. I expect it to be logging information _about_ the environment. Apparently it is actually setting up the logger, including information about the environment. Why not call `new_logger(env)`? As far as I can tell `log_environment` isn't necessary.

Reply:
As far as stdout vs. the log, I don't usually expect everything added to the log to also be printed to stdout.

Reply:
I originally had the `log_environment` method called `log_environment=` (to emphasize you were setting the environment), but it was giving me issues in trying to filter that down from the Module to the Class.  I tried doing `set_log_environment` next, but rubocop yelled about me about having methods with `set` in the name.

`new_logger` creates the logger and passes it back, but doesn't "save" the logger as a class variable.  ``new_logger` itself is meant to be more of a private thing.

I'll see if I can find a different name for this method to make it more clear.


As for the console vs the log file, I agree they don't necessarily need to say the same things, which is why their logging levels can be separate.  In fact, for the Ping method, and other "help" methods, it doesn't write to logs at all.  But during the harvesting-related jobs, I can't think of any statement that you would potentially put on the console that you wouldn't at least want to potentially put in a log file.  Having different log levels for the console and the file allow you to configure the verbosity separately for both.

Reply:
What is the rationale for the Module vs. adding the class methods? Is this architecture something you needed to do to work around Thor? This seems a bit complex for what we need to do here.

Reply:
What I expect to see in the Thor file is a log initialization like:

logger = SSTLogger.logger(environment or whatever other initialization information is needed)
logger.info('some information')

Usually the level of stdout information is controlled by a verbosity flag. If you want to tie stdout to log content, for simplicity's sake I would just have the use the same log level.

Reply:
The logger is intentionally a global thing - that way, every class is using the same logger without having to manually pass a logger into every initializer.  Since the "environment" in this app is not the "app environment" but rather the "target environment", I'm actually leaning towards not having environment-specific configuration at all; calling the CLI with `--environment integration` really isn't going to be any different than `--environment production` as far as the logging is concerned - the only difference is what DCS it calls and what Solr it writes to.

The "log_environment" is what is making this ugly.  If we ever have a need to do individual log setups we can re-evaluate at that point, but for now, I'll just refactor to have the logging configuration consistent.

Reply:
I'm all in favor of removing the environment.